### ISSUE TITLE
Use of dnsmasq.set_config injects unintentional text into the configuration file.

### ISSUE THREAD

**Issue by vernondcole** [saltstack/salt#34263]
### Description of Issue/Question

Operation of the dnsmasq.set_config module function adds lines to the configuration file which cause failure of dnsmasq.
### Setup

Salt has been installed in a Python virtual environment on a MacBook Pro running OS-X version 10.11.5, using git and a Homebrew current copy of Python, as suggested in [Installing Salt for development](https://docs.saltstack.com/en/develop/topics/development/hacking.html) as adapted for Mac OS. 

Note: the link on that page to the non-development installation for OS-x is not very helpful.

Dnsmasq was installed using Homebrew as suggested in [this blog](https://passingcuriosity.com/2013/dnsmasq-dev-osx/). 

In order for dnsmasq.version and dnsmasq.fullversion to work, I had to add dnsmasq to the PATH, which is not mentioned as a prerequisite, but ought to be:

`$ ln -s /usr/local/opt/dnsmasq/sbin/dnsmasq /usr/local/bin`
### Steps to Reproduce Issue

``` (bash)
$ sudo salt-call --local dnsmasq.set_config 'config_file=/usr/local/etc/dnsmasq.conf' domain=local address='/vern.test.local/8.8.4.4'
Password:
local:
    ----------
    __pub_fun:
        dnsmasq.set_config
    __pub_jid:
        20160623181002095141
    __pub_pid:
        8184
    __pub_tgt:
        salt-call
    address:
        /vern.test.local/8.8.4.4
    domain:
        local
$ tail /usr/local/etc/dnsmasq.conf

# Include all files in a directory which end in .conf
#conf-dir=/etc/dnsmasq.d/,*.conf
domain=local
__pub_pid=8184
__pub_fun=dnsmasq.set_config
__pub_jid=20160623181002095141
__pub_tgt=salt-call
address=/vern.test.local/8.8.4.4
$ sudo dnsmasq --no-daemon

dnsmasq: bad option at line 681 of /usr/local/etc/dnsmasq.conf
```
### Versions Report

```
$ salt --versions-report
Salt Version:
           Salt: 2016.3.0-784-g945b21b

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.11 (default, Jan 22 2016, 08:29:18)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.3
            ZMQ: 4.1.2

System Versions:
           dist:   
        machine: x86_64
        release: 15.5.0
         system: Darwin
        version: 10.11.5 x86_64
```

**Comment by Ch3LL** [comment]
@vernondcole I have been able to replicate this on 2015.5.10, 2015.8.1, 2015.8.10 and 2016.3. Also as a clarification question are you stating that we should include a prerequisite to add the dnsmasq command to PATH? In my opinion that should be included in dnsmasq's documentation as a setup step.

Just for anyone else coming by this issue if you remove the following from teh config file you will be able to get dnsmasq up and running:

```
__pub_pid=8184
__pub_fun=dnsmasq.set_config
__pub_jid=20160623181002095141
__pub_tgt=salt-call
```

**Comment by vernondcole** [comment]
@Ch3LL I agree that the PATH setup should be in the dnsmasq documentation. Perhaps it is, somewhere, but I did not see it. A short note in the Salt documentation would be helpful, too, since the module will not work as intended otherwise.

**Comment by vernondcole** [comment]
It seems that dnsmasq.py/set_config is now called with four keyword arguments it does not expect.  Here is a patch to ignore them. 

[34263.patch.txt](https://github.com/saltstack/salt/files/351165/34263.patch.txt)

Note to reviewer:  I think that the line just above my editing: 
`            if filename.endswith('#') and filename.endswith('#'):`
is oddly redundant.  I was tempted to fix it, but that was not what I went there to do.

**Comment by vernondcole** [comment]
@Ch3LL Is there also a label for "has patch"?  If so, it ought to be added to this issue.

**Comment by rallytime** [comment]
@vernondcole I have submitted #37390 to fix this issue, including a test so we won't regress here.

Thanks for your detective work! This was submitted to 2016.3 and will be available in 2016.3.5 release of Salt.
